Berikut ini adalah tahapan menganalisis sebuah data yang diperoleh dari Kaggle tentang kaitan antara luasan area toko dengan potensi kedatangan jumlah customer serta potensi penjualan yang terjadi di toko tersebut. 

Sumber data:
https://www.kaggle.com/datasets/surajjha101/stores-area-and-sales-data/code

Tahapan analisis data terdiri dari 2 langkah utama, yakni (1) membuat dataset & library serta (2) melakukan analisis data. 



#SET UP DATASET & LIBRARY

Pada tahap ini dilakukan 3 subtahap:
1. Quick look, yakni untuk melihat jenis data secara general serta jumlah baris dan kolomnya
2. Duplicate value, yakni untuk mengidentifikasi jika ada data ganda pada sumber data
3. Missing value, yakni untuk mengidentifikasi jika ada data hilang pada sumber data

In [17]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

## QUICK LOOK

In [31]:
df = pd.read_csv('Stores.csv')

In [19]:
df.shape

(896, 5)

In [21]:
df.head

<bound method NDFrame.head of      Store ID   Store_Area  Items_Available  Daily_Customer_Count  Store_Sales
0            1        1659             1961                   530        66490
1            2        1461             1752                   210        39820
2            3        1340             1609                   720        54010
3            4        1451             1748                   620        53730
4            5        1770             2111                   450        46620
..         ...         ...              ...                   ...          ...
891        892        1582             1910                  1080        66390
892        893        1387             1663                   850        82080
893        894        1200             1436                  1060        76440
894        895        1299             1560                   770        96610
895        896        1174             1429                  1110        54340

[896 rows x 5 columns

In [22]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 896 entries, 0 to 895
Data columns (total 5 columns):
 #   Column                Non-Null Count  Dtype
---  ------                --------------  -----
 0   Store ID              896 non-null    int64
 1   Store_Area            896 non-null    int64
 2   Items_Available       896 non-null    int64
 3   Daily_Customer_Count  896 non-null    int64
 4   Store_Sales           896 non-null    int64
dtypes: int64(5)
memory usage: 35.1 KB


Simpulan dari tahapan quicklook: 
- Semua data berjenis integer
- Sumber data terdiri dari 896 baris dan 5 kolom

##DUPLICATE VALUE

In [23]:
df.duplicated().sum()

0

Simpulan tahapan duplicate value:
Tidak ada data ganda. 

## MISSING VALUE

In [24]:
df.isnull().sum()

Store ID                0
Store_Area              0
Items_Available         0
Daily_Customer_Count    0
Store_Sales             0
dtype: int64

Simpulan tahapan missing value:
Tidak ada data hilang.

# DATA ANALYSIS

Pada tahap ini data mulai dianalisis untuk menghasilkan insight yang dapat bermanfaat untuk bisnis, yakni dengan melakukan analisis korelasi. 

In [33]:
df.max()

Store ID                   896
Store_Area                2229
Items_Available           2667
Daily_Customer_Count      1560
Store_Sales             116320
dtype: int64

In [34]:
df.corr()

Store ID   Store_Area  Items_Available  \
Store ID               1.000000   -0.058705        -0.055707   
Store_Area            -0.058705    1.000000         0.998891   
Items_Available       -0.055707    0.998891         1.000000   
Daily_Customer_Count  -0.010029   -0.041423        -0.040978   
Store_Sales            0.071486    0.097474         0.098849   

                      Daily_Customer_Count  Store_Sales  
Store ID                         -0.010029     0.071486  
Store_Area                       -0.041423     0.097474  
Items_Available                  -0.040978     0.098849  
Daily_Customer_Count              1.000000     0.008629  
Store_Sales                       0.008629     1.000000

#CONCLUSION

1. Adanya korelasi positif hampir sempurna antara store area dan item availability. Hal ini menunjukkan bahwa semakin besar area/luasan toko, maka item/barang yang diperjualbelikan di toko tersebut semakin banyak. 

2. Hampir tidak ada korelasi antara luasnya area toko dengan jumlah customer harian yang datang dan penjualan toko. Artinya, area toko yg semakin luas tidak menjamin peningkatan pelanggan dan penjualan toko. 